packages/ckeditor5-special-characters/src/ui/specialcharactersnavigationview.ts

```diff
@@ -7,81 +7,83 @@
+	declare public value: string;
+
 	/**
 	 * Creates an instance of the {@link module:special-characters/ui/specialcharactersnavigationview~SpecialCharactersNavigationView}
 	 * class.
 	 *
-	 * @param {module:utils/locale~Locale} locale The localization services instance.
-	 * @param {Map.<String, String>} groupNames The names of the character groups and their displayed labels.
+	 * @param locale The localization services instance.
+	 * @param groupNames The names of the character groups and their displayed labels.
 	 */
-	constructor( locale, groupNames ) {
+	constructor( locale: Locale, groupNames: GroupNames ) {
 		super( locale );
 
 		const t = locale.t;
+		const groupDefinitions = this._getCharacterGroupListItemDefinitions( groupNames );
 
+		this.set( 'value', groupDefinitions.first!.model.name as string );
 		this.set( 'class', 'ck-special-characters-navigation' );
-
-		/**
-		 * A dropdown that allows selecting a group of special characters to be displayed.
-		 *
-		 * @member {module:ui/dropdown/dropdownview~DropdownView}
-		 */
-		this.groupDropdownView = this._createGroupDropdown( groupNames );
+		this.groupDropdownView = this._createGroupDropdown( groupNames, groupDefinitions );
 		this.groupDropdownView.panelPosition = locale.uiLanguageDirection === 'rtl' ? 'se' : 'sw';
-
-		/**
-		 * @inheritDoc
-		 */
 		this.label = t( 'Special characters' );
-
-		/**
-		 * @inheritDoc
-		 */
 		this.children.add( this.groupDropdownView );
 	}
 
 	/**
 	 * Returns the name of the character group currently selected in the {@link #groupDropdownView}.
-	 *
-	 * @type {String}
 	 */
-	get currentGroupName() {
-		return this.groupDropdownView.value;
+	public get currentGroupName(): string {
+		return this.value;
 	}
 
 	/**
 	 * Focuses the character categories dropdown.
 	 */
-	focus() {
+	public focus(): void {
 		this.groupDropdownView.focus();
 	}
 
 	/**
 	 * Returns a dropdown that allows selecting character groups.
 	 *
-	 * @private
-	 * @param {Map.<String, String>} groupNames The names of the character groups and their displayed labels.
-	 * @returns {module:ui/dropdown/dropdownview~DropdownView}
+	 * @param groupNames The names of the character groups and their displayed labels.
+	 * @param groupDefinitions Definitions used in the character group dropdown to represent character groups.
 	 */
-	_createGroupDropdown( groupNames ) {
+	private _createGroupDropdown(
+		groupNames: GroupNames,
+		groupDefinitions: Collection<ListDropdownButtonDefinition>
+	): DropdownView {
 		const locale = this.locale;
-		const t = locale.t;
+		const t = locale!.t;
 		const dropdown = createDropdown( locale );
-		const groupDefinitions = this._getCharacterGroupListItemDefinitions( dropdown, groupNames );
-
-		dropdown.set( 'value', groupDefinitions.first.model.name );
 
-		dropdown.buttonView.bind( 'label' ).to( dropdown, 'value', value => groupNames.get( value ) );
+		dropdown.buttonView.bind( 'label' ).to( this, 'value', value => groupNames.get( value ) );
```

Comment:
Previously, the information about currently selected character group was stored in `DropdownView#value`. However, TypeScript wasn't happy about this, because the [DropdownView](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-ui/src/dropdown/dropdownview.ts#L78) doesn't have a `value` property.

I think it makes more sense to store this information in the property of this class anyway (see line 35). Could this have some side effects I'm not aware of besides the need to update the bindings? All tests are green (after a minor update) and I haven't found any problems when manually testing this plugin. @niegowski @arkflpc